generatorv2/README.md

```diff
@@ -0,0 +1,134 @@
+# Generator
+
+Last Updated: 2024/11/9
+
+The generator is a powerful tool that can be used to generate tests for exercises based on the canonical data.
+The generator is written in Ruby and is located in the `bin` directory.
+
+## How to use the generator
+
+### Things to do before running the generator
+
+Run `bundle install` to install the required libraries.
+Before running the generator you have to make sure a couple of files are in place.
+
+1. `tests.toml` file
+
+It is located under the `.meta` folder for each exercise.
+The toml file is used to configure which exercises are generated and which are not.
+Since the generator grabs all the data from the canonical data, so does this enable new tests that won't automatically be merged in.
+Instead so does new tests have to be added to the toml file before they show up in the test file.
+
+If there is a test that isn't needed or something that doesn't fit Ruby you can remove it from the toml file.
+By writing after the test name `include = false` and it will be skipped when generating the test file.
+
+2. `config.json` file, located in the root of the track
+
+The generator makes sure that the exercise is in the config.json so you need to add it there before running the generator.
+
+#### Things to note
+
+The script which grabs info from the toml file is quite sensitive, writing the toml file in an incorrect way can brick the generator.
+
+Here are some examples of how you should **NOT** work with the toml file.
+
+Make sure that the UUID is the only thing inside of `[UUID]`, if there is, for example, an extra space so would that break it.
+Here is an example
+
+```toml
+# This would break it since it is an extra space between UUID and `]`
+[1e22cceb-c5e4-4562-9afe-aef07ad1eaf4 ]
+# This would break it since it is an extra space between UUID and `[`
+[ 1e22cceb-c5e4-4562-9afe-aef07ad1eaf4]
+```
+
+The script won't care if you write `include = true` since if it sees the UUID it will always take it as long as `include = false` is not written.
+The script will not work if anything is misspelled, although the part which gets `include = false` doesn't care if it gets an extra space or not.
+
+**NOTE:**
+You are also **NOT** allowed to write `include = false` more than once after each UUID.
+Since that can lead to errors in the generator.
+
+Bad way:
+
+```toml
+[1e22cceb-c5e4-4562-9afe-aef07ad1eaf4]
+description = "basic"
+include = false
+include = false
+```
+
+Good way:
+
+```toml
+[1e22cceb-c5e4-4562-9afe-aef07ad1eaf4]
+description = "basic"
+include = false
+```
+
+### Template
+
+The generator uses a template file to generate the test file.
+The template is located under the `.meta` for each exercise.
+
+This template has to be manually written for each exercise.
+The goal although is to make it so that you only have to write the template once and then it will be able to be used to generate new tests.
```

Comment:
```suggestion
The goal is to make it so that you only have to write the template once and then it will be able to be used to generate new tests.
```